lib/Module/CMakeLists.txt

```diff
@@ -23,9 +23,14 @@ set(LLVM_COMPONENTS
   bitreader
   bitwriter
   codegen
+  instcombine
```

Comment:
While this works this is the wrong approach. `llvm_map_components_to_libnames()` is the way you used above will only give back `LLVMInstCombine`. The `LLVMInstCombine` library will have `INTERFACE` dependencies which should transitively cover all the dependent libraries. If it does not then there is a bug somewhere which we should report upstream.

Using `llvm_expand_dependencies()` is fine for a temporary fix but a comment should be added stating such and it should point at the relevant LLVM bugs

I have another project which uses LLVM 4.0 via CMake on my system so I'll try to take a look today at where the problem is and report it upstream if I can figure out what's going on.